model_api/python/openvino/model_api/models/classification.py

```diff
@@ -167,3 +226,81 @@ def addOrFindSoftmaxAndTopkOutputs(inference_adapter, topk):
 
 def sigmoid_numpy(x: np.ndarray):
     return 1.0 / (1.0 + np.exp(-x))
+
+
+def softmax_numpy(x: np.ndarray, eps: float = 1e-9):
+    x = np.exp(x - np.max(x))
+    return x / (np.sum(x) + eps)
+
+
+class GreedyLabelsResolver:
+    def __init__(self, hierarchical_config) -> None:
+        self.all_labels = list(
+            hierarchical_config["cls_heads_info"]["label_to_idx"].keys()
+        )
```

Comment:
That relies on an order of how `keys()` is iterated. I suggest this since you use it as map anyway
```suggestion
        self.idx_to_label = {
            idx: label
            for label, idx, in hierarchical_config["cls_heads_info"][
                "label_to_idx"
            ].items()
        }
        if len(self.idx_to_labeel) != len(
            hierarchical_config["cls_heads_info"]["label_to_idx"]
        ):
            raise RuntimeError(
                'All values in hierarchical_config["cls_heads_info"]["label_to_idx"] must be unique'
            )
```

Reply:
I actually need `label_to_idx` itself, just realized it